packages/kbn-zod-helpers/src/require_optional.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * Make any optional fields required, but add `| undefined` to their type.
+ *
+ * For example:
+ * type A = RequiredOptional<{ a?: string; b: number }>;
+ *
+ * will yield a type of:
+ * type A = { a: string | undefined; b: number; }
+ */
+export type RequiredOptional<T> = { [K in keyof T]-?: [T[K]] } extends infer U
+  ? U extends Record<keyof U, [unknown]>
+    ? { [K in keyof U]: U[K][0] }
+    : never
+  : never;
+
+/**
+ * This helper designed to be used with `z.transform` to make all optional fields required.
+ *
+ * @param schema Zod schema
+ * @returns The same schema but with all optional fields required.
+ */
+export const requireOptional = <T>(schema: T) => schema as RequiredOptional<T>;
```

Comment:
Why doesn't it change the schema itself though?

This can be a source of bugs if we or someone else needs to validate some input using a schema that relies on this helper.

We need to test it for request parameters of the CRUD endpoints.

Reply:
It doesn't make much sense to validate `undefined` values as they cannot be JSON serialized. `undefined` values are always removed from JSON thus this validation will always fail. We need it only to ensure that we do not forget to pass the values in our transformers and static type checking is good enough for that use case.

Reply:
@xcrzx ended up deleting this helper